File: python/number_of_laser_beams_in_a_bank.py
# no. 2125
class Solution:
    def count_lights(self, s):
        count = 0
        for c in s:
            if c == '1':
                count += 1
        return count

    def numberOfBeams(self, bank: List[str]) -> int:
        ans = 0
        top = 0
        down = 0
        for row in bank:
            n_row_lights = self.count_lights(row)
            if n_row_lights:
                if top == 0:
                    top = n_row_lights
                elif down == 0:
                    down = n_row_lights
                    ans += top * down
                    top = down
                    down = 0
        return ans

File: C++/number_of_laser_beams_in_a_bank.cpp
// no. 2125
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
  int countLights(string s) {
    int count = 0;
    for (int i = 0; i < s.length(); i++) {
      if (s[i] == '1') {
        count++;
      }
    }
    return count;
  }
  int numberOfBeams(vector<string> &bank) {
    int ans = 0;
    int top = 0;
    int down = 0;
    for (int i = 0; i < bank.size(); i++) {
      int n_row_lights = countLights(bank[i]);
      if (n_row_lights) {
        if (top == 0) {
          top = n_row_lights;
        } else if (down == 0) {
          down = n_row_lights;
          ans += down * top;
          top = down;
          down = 0;
        }
      }
    }
    return ans;
  }
};
